### PR TITLE
add the point evaluation precompile from EIP-4844

### PR DESCRIPTION
**Pull request by roberto-bayardo**

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -153,9 +154,23 @@ var PrecompiledContractsBLS = map[common.Address]PrecompiledContract{
...
+	common.BytesToAddress([]byte{2}):  &sha256hash{},
+	common.BytesToAddress([]byte{3}):  &ripemd160hash{},
+	common.BytesToAddress([]byte{4}):  &dataCopy{},
+	common.BytesToAddress([]byte{5}):  &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}):  &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}):  &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}):  &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}):  &blake2F{},
+	common.BytesToAddress([]byte{20}): &pointEvaluation{},
+}
```
Comment (0x00101010) [inline core/vm/contracts.go:168]: For my learning, what does this mean?

File: core/vm/testdata/precompiles/pointEvaluation.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[
+  {
+      "Input": "013c03613f6fc558fb7e61e75602241ed9a2f04e36d8670aadd286e71b5ca9cc420000000000000000000000000000000000000000000000000000000000000031e5a2356cbc2ef6a733eae8d54bf48719ae3d990017ca787c419c7d369f8e3c83fac17c3f237fc51f90e2c660eb202a438bc2025baded5cd193c1a018c5885bc9281ba704d5566082e851235c7be763b2a99adff965e0a121ee972ebc472d02944a74f5c6243e14052e105124b70bf65faf85ad3a494325e269fad097842cba",
+    "Expected": "000000000000000000000000000000000000000000000000000000000000100073eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001",
+    "Name": "pointEvaluation1",
+    "Gas": 50000,
+    "NoBenchmark": false
```
Comment (0x00101010) [inline core/vm/testdata/precompiles/pointEvaluation.json]: format better?

File: core/vm/contracts.go (hunk 1 above)
Comment (0x00101010) [reply core/vm/contracts.go:168]: Uhh, I think I understand now, these are list of pre-compiles for dank sharding fork?

**Review by 0x00101010 — APPROVED** [review]
Looks good other than the json formatting

File: core/vm/contracts.go (hunk 1 above)
Comment (roberto-bayardo) [reply core/vm/contracts.go:168]: Yes these are the list of pre-compiles that are allowed after danksharding is activated.

File: core/vm/testdata/precompiles/pointEvaluation.json (hunk 2 above)
Comment (roberto-bayardo) [reply core/vm/testdata/precompiles/pointEvaluation.json]: thx ran it through json-pretty-print

**Pull request merged**